apps/heft/src/plugins/CopyStaticAssetsPlugin.ts

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { LegacyAdapters, FileSystem, Terminal } from '@rushstack/node-core-library';
+import * as glob from 'glob';
+import * as globEscape from 'glob-escape';
+import * as path from 'path';
+import * as chokidar from 'chokidar';
+
+import { Async } from '../utilities/Async';
+import { performance } from 'perf_hooks';
+import { IPluginPackage } from '../pluginFramework/IPluginPackage';
+import { Build, HeftCompilation } from '../pluginFramework/HeftCompilation';
+import { HeftConfiguration } from '../configuration/HeftConfiguration';
+import { ICompilePhase, ICopyStaticAssetsConfiguration } from '../cli/actions/BuildAction';
+import { PrefixProxyTerminalProvider } from '../utilities/PrefixProxyTerminalProvider';
+
+const PLUGIN_NAME: string = 'CopyStaticAssetsPlugin';
+
+export const copyStaticAssetsPlugin: IPluginPackage = {
+  displayName: PLUGIN_NAME,
+  apply: (heftCompilation: HeftCompilation, heftConfiguration: HeftConfiguration) => {
+    heftCompilation.hooks.build.tap(PLUGIN_NAME, (build: Build) => {
+      build.hooks.compile.tap(PLUGIN_NAME, (compile: ICompilePhase) => {
+        const terminal: Terminal = new Terminal(
+          new PrefixProxyTerminalProvider(heftConfiguration.terminalProvider, '[copy-static-assets] ')
+        );
+
+        // For now - this may need to be revised later
+        compile.copyStaticAssetsConfiguration.sourceFolderName = 'src';
+        compile.copyStaticAssetsConfiguration.destinationFolders = ['lib'];
+
+        compile.hooks.run.tapPromise(PLUGIN_NAME, async () => {
+          const startTime: number = performance.now();
+
+          const configuration: ICopyStaticAssetsConfiguration = compile.copyStaticAssetsConfiguration;
+          if (!configuration.sourceFolderName) {
+            return;
+          }
+          const sourceFolderPath: string = path.join(
+            heftConfiguration.buildFolder,
+            configuration.sourceFolderName
+          );
+
+          async function expandGlobPattern(pattern: string): Promise<Set<string>> {
+            const results: string[] = await LegacyAdapters.convertCallbackToPromise(glob, pattern, {
+              cwd: sourceFolderPath,
+              nodir: true,
+              ignore: configuration.exclude
+            });
+
+            return new Set<string>(results);
+          }
+
+          let fileExtensionsGlobPattern: string | undefined = undefined;
+          if (
+            /* TODO: replace with "configuration.fileExtensions?.length" when ESLint can parse that */
```

Comment:
Coming very soon with PR https://github.com/microsoft/rushstack/pull/1936

Reply:
Cool - that's in its own commit that I can revert.

Reply:
@iclanton ESLint 7 is merged now -- give it a try